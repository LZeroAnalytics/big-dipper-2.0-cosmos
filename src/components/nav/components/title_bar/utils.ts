import Big from 'big.js';
import { formatNumber } from '@utils/format_token';

export const formatMarket = (data: {
  price: number;
  communityPool: TokenUnit;
  supply: TokenUnit;
  inflation: number;
  apr: number;
}) => {
  const marketCap = `$${formatNumber(
    (Number(data.supply.value) * data.price).toFixed(),
    2,
  )}`;

  return [
    {
      key: 'marketCap',
      data: marketCap,
    },
    {
      key: 'inflation',
      data: `${formatNumber(Big(data.inflation).times(100).toPrecision(), 0)}%`,
    },
    {
      key: 'apr',
      data: `${formatNumber(Big(data.apr).times(100).toPrecision(), 2)}%`,
    },
    {
      key: 'supply',
      data: `${formatNumber(
        data.supply.value,
        2,
      )} ${data.supply.displayDenom.toUpperCase()}`,
    },
    {
      key: 'communityPool',
      data: `${formatNumber(
        data.communityPool.value,
        2,
      )} ${data.communityPool.displayDenom.toUpperCase()}`,
    },
  ];
};
