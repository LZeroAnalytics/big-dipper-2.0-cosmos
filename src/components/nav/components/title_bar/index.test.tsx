import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer, { act } from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import axios from 'axios';
import TitleBar from '.';

// ==================================
// global setup
// ==================================
let component: renderer.ReactTestRenderer;

// ==================================
// mocks
// ==================================

const mockUseNavContext = {
  title: undefined,
  price: 0,
  marketCap: 0,
  inflation: 0,
  communityPool: 0,
};

// ==================================
// unit tests
// ==================================
describe('screen: Nav/TitleBar', () => {
  beforeEach(() => {
    component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <TitleBar />
        </MockTheme>
      </RecoilRoot>,
    );
  });

  const mAxiosResponse = {
    data: { coreum: { usd: 1 } },
  };

  jest.spyOn(axios, 'get').mockResolvedValueOnce(mAxiosResponse);

  it('it renders', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  act(() => {
    it('hook toggles correctly', () => {
      mockUseNavContext.title = 'Validators';
      component.update(
        <RecoilRoot>
          <MockTheme>
            <TitleBar />
          </MockTheme>
        </RecoilRoot>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

afterEach(() => {
  mockUseNavContext.title = undefined;
  jest.clearAllMocks();
});
