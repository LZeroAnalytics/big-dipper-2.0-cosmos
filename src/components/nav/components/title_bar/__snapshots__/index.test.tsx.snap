// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Nav/TitleBar hook toggles correctly 1`] = `
<div
  className="makeStyles-root"
>
  <div
    className="makeStyles-content"
  >
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:marketCap
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        $0
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:inflation
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0%
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:apr
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0%
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:supply
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0 
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:communityPool
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0 
      </p>
    </div>
  </div>
</div>
`;

exports[`screen: Nav/TitleBar it renders 1`] = `
<div
  className="makeStyles-root"
>
  <div
    className="makeStyles-content"
  >
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:marketCap
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        $0
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:inflation
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0%
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:apr
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0%
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:supply
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0 
      </p>
    </div>
    <div
      className="makeStyles-item"
    >
      <p
        className="MuiTypography-root label MuiTypography-body1"
      >
        common:communityPool
      </p>
      <p
        className="MuiTypography-root data MuiTypography-body1"
      >
        0 
      </p>
    </div>
  </div>
</div>
`;
